packer/provisioner.go

```diff
@@ -145,15 +145,67 @@ func (p *PausedProvisioner) Prepare(raws ...interface{}) error {
 
 func (p *PausedProvisioner) Provision(ctx context.Context, ui packersdk.Ui, comm packersdk.Communicator, generatedData map[string]interface{}) error {
 
-	// Use a select to determine if we get cancelled during the wait
 	ui.Say(fmt.Sprintf("Pausing %s before the next provisioner...", p.PauseBefore))
+
+	var err error
+	switch ui.(type) {
+	case *MachineReadableUi:
+		err = p.pausingNoUpdate(ctx)
+	default:
+		minimumTimeForPauseUpdate := float64(10)
+		if p.PauseBefore.Seconds() < minimumTimeForPauseUpdate {
+			err = p.pausingNoUpdate(ctx)
+		} else {
+			err = p.pausingWithUpdates(ctx, ui)
+		}
+	}
+	if err != nil {
+		return err
+	}
+
+	return p.Provisioner.Provision(ctx, ui, comm, generatedData)
+}
+
+func (p *PausedProvisioner) pausingNoUpdate(ctx context.Context) error {
 	select {
 	case <-time.After(p.PauseBefore):
 	case <-ctx.Done():
 		return ctx.Err()
 	}
+	return nil
+}
 
-	return p.Provisioner.Provision(ctx, ui, comm, generatedData)
+func (p *PausedProvisioner) pausingWithUpdates(ctx context.Context, ui packersdk.Ui) error {
+	updateTime := 10
+	ticker := time.NewTicker(time.Duration(updateTime) * time.Second)
+	defer ticker.Stop()
+	totalTime := p.PauseBefore.Seconds()
+
+	pausingCtx, cancel := context.WithTimeout(context.Background(), p.PauseBefore)
```

Comment:
Here, if we use the passed in `ctx`, then this context will also end if someone/something cancels the top context.
```suggestion
	pausingCtx, cancel := context.WithTimeout(ctx, p.PauseBefore)
```